integration/kotlinx-coroutines-play-services/src/Tasks.kt

```diff
@@ -95,13 +133,18 @@ public suspend fun <T> Task<T>.await(): T {
 
     return suspendCancellableCoroutine { cont ->
         addOnCompleteListener {
-            val e = exception
+            val e = it.exception
             if (e == null) {
                 @Suppress("UNCHECKED_CAST")
-                if (isCanceled) cont.cancel() else cont.resume(result as T)
+                if (it.isCanceled) cont.cancel() else cont.resume(it.result as T)
             } else {
                 cont.resumeWithException(e)
             }
         }
+
+        if (cancellationTokenSource == null) return@suspendCancellableCoroutine
+        cont.invokeOnCancellation {
+            cancellationTokenSource.cancel()
+        }
```

Comment:
Minor, opinion: I know that you like things de-indented, but to me, this looks just unnatural.

Reply:
Sure, reverted the condition